compiler-rt/lib/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cpp

```diff
@@ -37,25 +37,28 @@
 # include <asm/ptrace.h>
 #endif
 #include <sys/user.h>  // for user_regs_struct
-#if SANITIZER_ANDROID && SANITIZER_MIPS
-# include <asm/reg.h>  // for mips SP register in sys/user.h
-#endif
-#include <sys/wait.h> // for signal-related stuff
-
-#ifdef sa_handler
-# undef sa_handler
-#endif
-
-#ifdef sa_sigaction
-# undef sa_sigaction
-#endif
-
-#include "sanitizer_common.h"
-#include "sanitizer_flags.h"
-#include "sanitizer_libc.h"
-#include "sanitizer_linux.h"
-#include "sanitizer_mutex.h"
-#include "sanitizer_placement_new.h"
+#  if SANITIZER_MIPS
+// clang-format off
```

Comment:
Why do we need `clang-format off`

Reply:
Otherwise it would reorder the includes alphabetically and put `regs.h` before `sgidefs.h`, which won't work because `regs.h` depends on definitions in `sgidefs.h`.

Reply:
Please comment in the code.

Reply:
I guess those '// for' comments for that but something more direct"
"<asm/sgidefs.h> must be included before <asm/reg.h>"

Reply:
Should be fixed now. I also moved the formatting to a separate commit.